R/Design-class.R

```diff
@@ -989,3 +989,208 @@ DesignGrouped <- function(model,
 #' @note Typically, end-users will not use the `.DefaultDesignGrouped()` function.
 #' @export
 .DefaultDesignGrouped <- .DesignGrouped
+
+# RuleDesignOrdinal ----
+
+## class ----
+
+#' `RuleDesignOrdinal`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' [`RuleDesignOrdinal`] is the class for rule-based designs. The difference between
+#' this class and the [`DesignOrdinal`] class is that [`RuleDesignOrdinal`]
+#' does not contain `model`, `stopping` and `increments` slots.
+#'
+#' @slot next_best (`NextBest`)\cr how to find the next best dose.
+#' @slot cohort_size (`CohortSize`)\cr rules for the cohort sizes.
+#' @slot data (`Data`)\cr specifies dose grid, any previous data, etc.
+#' @slot starting_dose (`number`)\cr the starting dose, it must lie on the dose
+#'   grid in `data`.
+#'
+#' @aliases RuleDesignOrdinal
+#' @export
+#'
+.RuleDesignOrdinal <- setClass(
+  Class = "RuleDesignOrdinal",
+  slots = c(
+    next_best = "NextBestOrdinal",
+    cohort_size = "CohortSizeOrdinal",
+    data = "DataOrdinal",
+    starting_dose = "numeric"
+  ),
+  prototype = prototype(
+    next_best = .NextBestOrdinal(),
+    cohort_size = CohortSizeOrdinal(1L, CohortSizeConst(3L)),
+    data = DataOrdinal(doseGrid = 1:3),
+    starting_dose = 1
+  ),
+  contains = "CrmPackClass",
+  validity = v_rule_design_ordinal
+)
+
+## constructor ----
+
+#' @rdname RuleDesignOrdinal-class
+#'
+#' @param next_best (`NextBestOrdinal`)\cr see slot definition.
+#' @param cohort_size (`CohortSizeOrdinal`)\cr see slot definition.
+#' @param data (`DataOrdinal`)\cr see slot definition.
+#' @param starting_dose (`number`)\cr see slot definition.
+#'
+#' @export
+#' @example examples/Design-class-RuleDesignOrdinal.R
+#'
+RuleDesignOrdinal <- function(
+    next_best,
+    cohort_size,
+    data,
+    starting_dose) {
+  new(
+    "RuleDesignOrdinal",
+    next_best = next_best,
+    cohort_size = cohort_size,
+    data = data,
+    starting_dose = as.numeric(starting_dose)
+  )
+}
+
+#' @rdname RuleDesignOrdinal-class
+#' @note Typically, end users will not use the `.DefaultRuleDesignOrdinal()` function.
+#' @export
+
+.DefaultRuleDesignOrdinal <- function() {
+  RuleDesignOrdinal(
+    next_best = CohortSizeOrdinal(1L, NextBestMTD(target = 0.25)),
+    cohort_size = CohortSizeOrdinal(1L, CohortSizeConst(size = 3L)),
+    data = DataOrdinal(doseGrid = c(5, 10, 15, 25, 35, 50, 80)),
+    starting_dose = 5
+  )
+}
+
+# DesignOrdinal ----
+
+## class ----
+
+#' `DesignOrdinal`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' [`DesignOrdinal`] is the class for rule-based ordinal designs. The difference
+#' between this class and its parent [`RuleDesignOrdinal`] class is that the
+#'  [`DesignOrdinal`] class contains additional `model`, `stopping` and
+#'  `increments` slots.
```

Comment:
also mention `pl_cohort_size` (or move it to the rule design class)

Reply:
I've kept `pl_cohort_size` in `DesignOrdinal` for consistency with `Design`.